lib/libc/Kconfig

```diff
@@ -35,8 +35,8 @@ config NEWLIB_LIBC_SUPPORTED
 config PICOLIBC_SUPPORTED
 	bool
 	depends on !NATIVE_APPLICATION
-	depends on ("$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr") || (NATIVE_LIBRARY)
```

Comment:
These conditionals are super twitchy, especially the second one. I'm unsure precisely how this works today, but for glibcxx_libcpp to work, both it and picolibc must come from the toolchain, and this condition is somehow doing that.

Reply:
Hm... `CONFIG_PICOLIBC_USE_MODULE` does `depends on !GLIBCXX_LIBCPP`. So if `GLIBCXX_LIBCPP` is enabled, building the module is not an option. Though `GLIBCXX_LIBCPP` should not be gating `CONFIG_PICOLIBC_SUPPORTED`?...

Reply:
yeah, there have been incremental changes to these configuration conditions leaving me very unsure of what the right values should be now.

Reply:
the essential requirement is that the C library and libstdc++ must both be provided by the toolchain as libstdc++ must be built atop the associated C library. The existing conditions try to limit picolibc to environments which are known to work. For C++, the only known working configuration is to have the Zephyr SDK provide both libstdc++ and picolibc; adding more combinations here when toolchains provide both picolibc and the standard C++ library.

Reply:
The kconfig combinations in this PR should keep that behavior and not allowing toolchains without `TOOLCHAIN_HAS_PICOLIBC` to use GNU C++ STL.